infrastructure/adapters/persistence/src/main/java/md/maib/retail/infrastructure/persistence/EffectRecord.java

```diff
@@ -0,0 +1,19 @@
+package md.maib.retail.infrastructure.persistence;
+
+import lombok.AllArgsConstructor;
+import md.maib.retail.model.effects.Effect;
+import md.maib.retail.model.effects.LoyaltyEffectType;
+
+import java.util.UUID;
+@AllArgsConstructor
+public class EffectRecord {
+    private UUID loyaltyEventTypeId;
+
+    private String value;
+
+    public static Effect toEffect(EffectRecord effectRecord){
+        LoyaltyEffectTypesAdapter loyaltyEffectTypesAdapter=new LoyaltyEffectTypesAdapter();
```

Comment:
Why create instance of adapter for each function call?
You can inject it in SpringDataJpaProjectRepositoryAdapter and later use it